### ISSUE TITLE
0.56.3 `terraform plan` runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by tkmcc** [Snowflake-Labs/terraform-provider-snowflake#1531]
**Provider Version**

0.56.3

**Terraform Version**

1.3.7

**Describe the bug**

Running `terraform plan` crashes while refreshing state, with the following error:
```
Stack trace from the terraform-provider-snowflake_v0.56.3 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x14e2afe]

goroutine 3936 [running]:
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Set).hash(0xc00033b1a0?, {0x1cfb020?, 0xc000a51230?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/set.go:218 +0x1e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Set).Contains(...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/set.go:86
github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/resources.readGenericGrant(0xc000123680, {0x1ea9600?, 0xc00053add0?}, 0x1?, {0x2110ce0, 0xc0009cf200}, 0x1, 0x2?)
	github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/resources/grant_helpers.go:283 +0x89d
github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/resources.ReadSchemaGrant(0xc000123680, {0x1ea9600, 0xc00053add0})
	github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/resources/schema_grant.go:253 +0x498
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x210f3a0?, {0x210f3a0?, 0xc000a857d0?}, 0xd?, {0x1ea9600?, 0xc00053add0?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:712 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc00041d260, {0x210f3a0, 0xc000a857d0}, 0xc000beb860, {0x1ea9600, 0xc00053add0})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:1015 +0x585
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc00040e408, {0x210f3a0?, 0xc000a856b0?}, 0xc0009cef40)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/grpc_provider.go:613 +0x4a5
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc0002e6b40, {0x210f3a0?, 0xc000a850e0?}, 0xc000d5b800)
	github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/tf5server/server.go:748 +0x4b1
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x1e763a0?, 0xc0002e6b40}, {0x210f3a0, 0xc000a850e0}, 0xc000161420, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:349 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000414000, {0x2114ec0, 0xc000602820}, 0xc0008045a0, 0xc0006370b0, 0x2893b90, 0x0)
	google.golang.org/grpc@v1.50.1/server.go:1340 +0xd23
google.golang.org/grpc.(*Server).handleStream(0xc000414000, {0x2114ec0, 0xc000602820}, 0xc0008045a0, 0x0)
	google.golang.org/grpc@v1.50.1/server.go:1713 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.50.1/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.50.1/server.go:963 +0x28a

Error: The terraform-provider-snowflake_v0.56.3 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

**Expected behavior**

plan succeeds

**Comment by sfc-gh-asrivastava** [comment]
@sfc-gh-swinkler : This is to bring your kind attention .Can you please provide any ETA for the above issue? TIA

**Comment by sfc-gh-swinkler** [comment]
This has been resolved in 0.56.4
